### PR TITLE
Feature/29 merge slackbot fn to backend

### PR DESCRIPTION
**Pull request by wayangalihpratama**
### TODO / DONE

---

- [x] Update docker compose backend, add slackbot env
- [x] Update docker test, add slackbot env
- [x] Remove slackbot container from docker
- [x] Move slackbot into backend and add test
- [x] Move slackbot routes into backend and add test
- [x] Delete slackbot folder

---

- [x] Remove twilibot/slacbot queue routes
- [x] Revert twilibot/slackbot queue implementation
- [x] Add new consume queue implementation based on `platform` (check software design doc)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1207713342872859

### CODE REVIEW

**Review by kjkoster — CHANGES_REQUESTED** [review]
Hi Galih, this PR seems to be a mix of good changes and ones that feel a tad unfinished to me Can you please tell me what your intention is? Do you propose to merge this into master or did you expect me to do an intermediate review?

File: backend/clients/twilio_client.py [hunk 1]
```diff
@@ -57,17 +55,17 @@ def chunk_text_by_paragraphs(self, text: str, max_length: int) -> list[str]:
...
         try:
             queue_message = json.loads(body)
-            text = queue_message["text"]
-            phone = queue_message["to"]["phone"]
+            text = queue_message.get("body")
+            conversation_envelope = queue_message.get(
+                "conversation_envelope", {})
+            phone = conversation_envelope.get("client_phone_number")
+            if not phone:
+                phone = conversation_envelope.get("user_phone_number")
```
[the remark this answers is not in the document]
Comment (dedenbangkit) [reply backend/clients/twilio_client.py:65]: Yes, please write comment / notes @wayangalihpratama.

You can also shorten the code to:
```python
phone = conversation.get("client_phone_number", conversation.get("user_phone_number"))
```

File: backend/clients/twilio_client.py [hunk 2]
```diff
@@ -105,31 +102,31 @@ def validate_and_format_phone_number(self, phone_number: str) -> str:
             raise ValidationError(f"Phone number validation error: {e}")
 
     def format_to_queue_message(self, values: dict) -> str:
-        """
-        Consume messages from Twilio then format into queue message.
-        """
-        iso_timestamp = datetime.now().isoformat()
         try:
+            iso_timestamp = datetime.now().isoformat()
```
[the remark this answers is not in the document]
Comment (dedenbangkit) [reply backend/clients/twilio_client.py:106]:
```python
datetime.now(timezone.utc).isoformat()
```

File: backend/clients/twilio_client.py [hunk 3]
```diff
@@ -105,31 +102,31 @@ def validate_and_format_phone_number(self, phone_number: str) -> str:
...
+            media = []
             num_media = values.get('NumMedia', 0)
             for i in range(num_media):
                 media_url = values.get(f'MediaUrl{i}', '')
                 if media_url:
-                    queue_message['media'].append(media_url)
+                    media.append(media_url)
+            queue_message = queue_message_util.create_queue_message(
+                message_id=values["MessageSid"],
+                conversation_id="__CHANGEME__",
```
[the remark this answers is not in the document]
Comment (wayangalihpratama) [reply backend/clients/twilio_client.py:120]: When we receive a message from Twilio, we will format the message into queue format then put that message into queue. The incoming message will have `MessageSid` as our message_id in queue, and should we set and use `uuid` as a conversation_id here?

File: backend/routes/user_routes.py [hunk 4]
```diff
@@ -61,17 +59,19 @@ async def verify_login_code(
     return {"token": login_token}
 
 
-async def send_whatsapp_message(phone_number: int, login_token: str):
+async def send_to_twilio(phone_number: int, login_token: str):
```
[the remark this answers is not in the document]
Comment (dedenbangkit) [reply backend/routes/user_routes.py]: https://github.com/akvo/rag-doll/pull/30/commits/6a40d5c3fa479ff38a64e92ada2bd9ff590bfaf6

File: frontend/src/components/chats/ChatWindow.js [hunk 5]
```diff
@@ -70,14 +70,28 @@ const ChatWindow = () => {
...
+        };
+      }
       const chatPayload = createQueueMessage({
-        messageId: uuidv4(),
-        conversationId: uuidv4(),
-        userPhoneNumber: "+628123456789",
-        clientPhoneNumber: "+628223456789",
-        sender: "CLIENT",
+        ...chatBreakdown,
+        client_phone_number: "_CHANGEME_", // change to your phone number to test
```
[the remark this answers is not in the document]
Comment (wayangalihpratama) [reply frontend/src/components/chats/ChatWindow.js]: When should be the conversation ID defined?

**Review by dedenbangkit — APPROVED** [review]

File: backend/routes/slack_routes.py [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+slackbot_client = SlackBotClient()
+slack_app = slackbot_client.slack_app
+
+router = APIRouter()
+security = HTTPBearer()
+
+RABBITMQ_QUEUE_SLACKBOT_REPLIES = os.getenv("RABBITMQ_QUEUE_SLACKBOT_REPLIES")
```
Comment (kjkoster) [inline backend/routes/slack_routes.py:19]: This can go.

File: frontend/src/components/chats/ChatWindow.js [hunk 7]
```diff
@@ -87,7 +100,7 @@ const ChatWindow = () => {
 
   const renderChatMessages = () => {
     return chats.map((c, ci) => {
-      if (c?.conversation_envelope?.sender === "CLIENT") {
+      if (c?.conversation_envelope?.sender_role === "user") {
```
Comment (kjkoster) [inline frontend/src/components/chats/ChatWindow.js:103]: Please make that an enum.

**Review by kjkoster — CHANGES_REQUESTED** [review]

**Comment by wayangalihpratama** [comment]
I will remove the conversation ID in the new PR related to HTTP session.

File: backend/tests/conftest.py [hunk 8]
```diff
@@ -108,3 +109,28 @@ def override_get_db():
...
+    async def initialize(self):
+        pass
+
+    async def disconnect(self):
+        pass
+
+    async def consume(self, queue_name, routing_key, callback):
+        pass
+
+    async def producer(self, body, routing_key, reply_to):
```
Comment (kjkoster) [inline backend/tests/conftest.py:124]: Is `reply_to` needed?

**Review by kjkoster — APPROVED** [review]

**Pull request merged**